src/llvm_backend_debug.cpp

```diff
@@ -1301,3 +1301,88 @@ gb_internal void add_debug_info_for_global_constant_from_entity(lbGenerator *gen
 		}
 	}
 }
+
+// TODO(tf2spi) ... *sigh* I hate LLVM so much.
+// LLVM-C doesn't have an API to emit DILabel yet, so this is not exported in the dll.
+// For POSIX, we can temporarily use the C++ APIs, but we're out of luck for Windows
+// until the DLL gets updated with definitions like LLVMDIBuilderCreateLabel.
+// These might be the ~80 most disgusting lines in this entire compiler.
+#ifndef _WIN32
+// Don't you just love a good ol' mangled name? I know I do.
+// Mangled name of "llvm::DIBuilder::createLabel(DIScope, StringRef, DIFile, unsigned, bool)"
+extern "C" LLVMMetadataRef _ZN4llvm9DIBuilder11createLabelEPNS_7DIScopeENS_9StringRefEPNS_6DIFileEjb(
+	LLVMDIBuilderRef Builder,
+	LLVMMetadataRef Scope,
+	String Name, // Not technically proper to do, but it's ABI compatible for now...
+	LLVMMetadataRef File,
+	unsigned int LineNo,
+	bool AlwaysPreserve);
+// Mangled name of "llvm::DIBuilder::insertLabel(DILabel, DILocation, BasicBlock)"
+extern "C" LLVMMetadataRef _ZN4llvm9DIBuilder11insertLabelEPNS_7DILabelEPKNS_10DILocationEPNS_10BasicBlockE(
+	LLVMDIBuilderRef Builder,
+	LLVMMetadataRef Label,
+	LLVMMetadataRef Location,
+	LLVMBasicBlockRef Block);
+#endif
+gb_internal void lb_add_debug_label(lbProcedure *p, Ast *label, lbBlock *target) {
+	// TODO(tf2spi): Work with gingerBill to see if we can patch LLVM-C
+	//               to export DILabel functions for the C API on Windows
+	//               or to determine whether it is worthwhile to do this.
+	//               Until then, unimplemented, so just return.
```

Comment:
I do not want to change the LLVM-C API. It has to be whatever the official one is.

Reply:
In retrospect, I now realize this is the more wise approach. Better to ask [llvm-project](https://github.com/llvm/llvm-project) to add a new C API that we can eventually use. [This ORCv2/LLJIT issue](https://github.com/llvm/llvm-project/issues/70666) is a fairly recent example of them adding the API fairly quickly when they detect a valid use case.

Reply:
[My PR adding DebugInfo labels](https://github.com/llvm/llvm-project/pull/112840) got added recently. Whenever it's convenient and useful to update the ``LLVM-C`` DLL for Windows, I can add a case for Windows to use the new functions.